### ISSUE TITLE
Is it possible to replace EMA of feature norm calculation with a torch.nn.BatchNorm layer?

### ISSUE THREAD

**Issue by JokaHD** [mk-minchul/AdaFace#2]
Hi,

I'm trying to implement AdaFace using MXNet and wondering if it is reasonable to replace EMA of feature norm calculation  with a BatchNorm layer (without learnable parameters).

This is to say, can I replace this part
https://github.com/mk-minchul/AdaFace/blob/79af07a1ea17cc2022af3ee04fd58485d8ee1187/head.py#L75-L81
with this in constructor
```python
self.norm_layer = torch.nn.BatchNorm1d(1, eps=self.eps, momentum=self.t_alpha, affine=False)
```
and this in forward()?
```python
margin_scaler = self.norm_layer(safe_norms)
```

I'm not sure if `with torch.no_grad()` should be kept or there is something I haven't noticed.
Thank you.

**Comment by mk-minchul** [comment]
Yes, replacing the manual updates of batch statistics with the batchnorm layer is equivalent and a simpler way to replace it. Thanks for the suggestion.

**Comment by mk-minchul** [comment]
torch no grad can be eliminated if you make sure that no gradient flows through the norm (safe_norms)

**Comment by JokaHD** [comment]
OK, thanks for the reply.
